templates/cdap-ui-deployment.yaml

```diff
@@ -14,55 +14,52 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{.Name}}
-  namespace: {{.Master.Namespace}}
-{{if .Labels}}
+  name: {{.Base.Name}}
+  namespace: {{.Base.Namespace}}
+{{if .Base.Labels}}
   labels:
-    {{range $k,$v := .Labels}}
-    {{$k}}: '{{$v}}'
+    {{range $k,$v := .Base.Labels}}
+    {{$k}}: {{$v}}
     {{end}}
 {{end}}
 spec:
   selector:
     matchLabels:
-      {{range $k,$v := .Labels}}
-      {{$k}}: '{{$v}}'
+      {{range $k,$v := .Base.Labels}}
+      {{$k}}: {{$v}}
       {{end}}
-{{if .Replicas}}
-  replicas: {{.Replicas}}
-{{else}}
-  replicas: 1
-{{end}}
+  replicas: {{.Base.Replicas}}
   template:
     metadata:
       labels:
-        {{range $k,$v := .Labels }}
-        {{$k}}: '{{$v}}'
+        {{range $k,$v := .Base.Labels }}
+        {{$k}}: {{$v}}
         {{end}}
     spec:
-    {{if .ServiceAccountName}}
-      serviceAccountName: {{.ServiceAccountName}}
+    {{if .Base.ServiceAccountName}}
+      serviceAccountName: {{.Base.ServiceAccountName}}
     {{end}}
-    {{if .Service.NodeSelector}}
+    {{if .Base.NodeSelector}}
       nodeSelector:
-        {{range $k,$v := .Service.NodeSelector}}
+        {{range $k,$v := .Base.NodeSelector}}
         {{$k}}: '{{$v}}'
         {{end}}
     {{end}}
-    {{if .Service.RuntimeClassName}}
+    {{if .Base.RuntimeClassName}}
       runtimeClassName: {{.Service.RuntimeClassName}}
     {{end}}
-    {{if .Service.PriorityClassName}}
+    {{if .Base.PriorityClassName}}
       priorityClassName: {{.Service.PriorityClassName}}
     {{end}}
+    {{range $c := .Containers}}
       containers:
-        - image: {{.Master.Status.UserInterfaceImageToUse}}
+        - name: {{$c.Name}}
+          image: {{$c.Image}}
           workingDir: "/opt/cdap/ui"
```

Comment:
is working dir same for all containers (non ui ones)

Reply:
Yes. For this specific template, it is used by our UI service, which will be a normal single-container pod.